### PR TITLE
Make use of secure port when accessing Kubelet API

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?

This PR switches Metricbeat k8s manifests and docs to point to Kubelet secure port over https instead of the insecure port.

## Why is it important?

Insecure port of Kubelet (10255/TCP) is now less common and discouraged and also in most cases it is not enabled by default (requiring to restart `kubelet` with `--read-only-port` flag)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~

## Related issues

- Closes https://github.com/elastic/beats/issues/14420

### CODE REVIEW

File: deploy/kubernetes/metricbeat-kubernetes.yaml [hunk 1]
```diff
@@ -76,11 +76,11 @@ data:
         - volume
       period: 10s
       host: ${NODE_NAME}
-      hosts: ["localhost:10255"]
-      # If using Red Hat OpenShift remove the previous hosts entry and 
+      hosts: ["https://${HOSTNAME}:10250"]
+      bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+      ssl.verification_mode: "none"
+      # If using Red Hat OpenShift remove the previous hosts entry and
       # uncomment these settings:
```
Comment (exekias) [inline deploy/kubernetes/metricbeat-kubernetes.yaml]: this comment needs rewording, right?

**Comment by exekias** [comment]
this will need a changelog

File: deploy/kubernetes/metricbeat-kubernetes.yaml (hunk 1 above)
Comment (ChrsMark) [reply deploy/kubernetes/metricbeat-kubernetes.yaml]: 👍

**Comment by ChrsMark** [comment]
Also these changes might apply to [ECK manifests](https://github.com/elastic/cloud-on-k8s/blob/35968f3289e2b265916d264d25fff3e7ce56edb2/config/recipes/beats/3_metricbeat-kubernetes.yaml#L79) and [helm-chart1](https://github.com/elastic/helm-charts/blob/a6de6afae37a17c4385d337f2a6602fb89002e0e/metricbeat/values.yaml#L16) [helm-chart2](https://github.com/elastic/helm-charts/blob/0e65e806594ae4657c8371c0f0027d5609ee559e/metricbeat/examples/security/values.yaml#L13) maybe?

**Comment by exekias** [comment]
> Also these changes might apply to [ECK manifests](https://github.com/elastic/cloud-on-k8s/blob/35968f3289e2b265916d264d25fff3e7ce56edb2/config/recipes/beats/3_metricbeat-kubernetes.yaml#L79) and [helm-chart1](https://github.com/elastic/helm-charts/blob/a6de6afae37a17c4385d337f2a6602fb89002e0e/metricbeat/values.yaml#L16) [helm-chart2](https://github.com/elastic/helm-charts/blob/0e65e806594ae4657c8371c0f0027d5609ee559e/metricbeat/examples/security/values.yaml#L13) maybe?

Yes! good catch

**Comment by exekias** [comment]
can you also backport this to 7.6? it's docs only so it should not really affect the release

**Pull request merged**